lua/telescope/builtin/internal.lua

```diff
@@ -87,6 +87,71 @@ internal.planets = function(opts)
   }:find()
 end
 
+internal.symbols = function(opts)
+  -- TODO(conni2461): Is their a better way to get the extension path?
+  local sourced_file = require('plenary.debug_utils').sourced_filepath()
+  local base_directory = vim.fn.fnamemodify(sourced_file, ":h:h:h:h:h")
+  local extension_path = vim.fn.globpath(base_directory .. '/telescope-symbols.nvim/data/', '*', true, true)
+
+  local globbed_files = vim.fn.globpath(vim.fn.stdpath('data') .. '/telescope-sources/', '*', true, true)
+
+  local files = vim.tbl_extend('force', extension_path, globbed_files)
+  if table.getn(files) == 0 then
+    print("No sources found! Check out https://github.com/nvim-telescope/telescope-symbols.nvim " ..
+          "for some prebuild symbols or how to create you own symbol source.")
+    return
+  end
+
+  local sources = {}
+  if opts.sources then
+    for _, v in ipairs(files) do
+      for _, s in ipairs(opts.sources) do
+        if v:find(s) then
+          table.insert(sources, v)
+        end
+      end
+    end
+  else
+    sources = files
+  end
+
+  local results = {}
+  for _, v in ipairs(sources) do
+    local fd = assert(vim.loop.fs_open(v, "r", 438))
+    local stat = assert(vim.loop.fs_fstat(fd))
+    local data = assert(vim.loop.fs_read(fd, stat.size, 0))
+    assert(vim.loop.fs_close(fd))
+    data = vim.fn.json_decode(data)
+    for _, v in ipairs(data) do
+      table.insert(results, v)
+    end
+  end
+
+  pickers.new(opts, {
+    prompt_title = 'Telescope Builtin',
+    finder    = finders.new_table {
+      results     = results,
+      entry_maker = function(entry)
+        return {
+          value = entry,
+          ordinal = entry[1] .. ' ' .. entry[2],
+          display = entry[1] .. ' ' .. entry[2],
+        }
+      end
+    },
+    sorter = conf.generic_sorter(opts),
+    attach_mappings = function(prompt_bufnr)
+      actions.goto_file_selection_edit:replace(function()
+        local selection = actions.get_selected_entry()
+        actions.close(prompt_bufnr)
+        local symbol = selection.value[1]
+        vim.api.nvim_put({symbol}, '', true, true)
+      end)
```

Comment:
Let's make this it's own action.

Then you can do `actions.goto_file_selection_edit:replace(actions.insert_symbol)`

Reply:
Wait a sec i am currently explaining everything. But yeah i can do that